include/exadg/incompressible_navier_stokes/user_interface/viscosity_model_data.h

```diff
@@ -0,0 +1,131 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2021 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_EXADG_INCOMPRESSIBLE_NAVIER_STOKES_USER_INTERFACE_VISCOSITY_MODEL_DATA_H_
+#define INCLUDE_EXADG_INCOMPRESSIBLE_NAVIER_STOKES_USER_INTERFACE_VISCOSITY_MODEL_DATA_H_
+
+namespace ExaDG
+{
+namespace IncNS
+{
+/*
+ *  Turbulence model data.
+ */
+struct TurbulenceModelData
+{
+  TurbulenceModelData()
+    : turbulence_model(TurbulenceEddyViscosityModel::Undefined), is_active(false), constant(0.0)
+  {
+  }
+
+  TurbulenceEddyViscosityModel turbulence_model;
+  bool                         is_active;
+  double                       constant; // model constant
+
+  void
+  check() const
+  {
+    AssertThrow(is_active, dealii::ExcMessage("Turbulence model is inactive."));
+    AssertThrow(constant > 1e-20, dealii::ExcMessage("Parameter must be greater than zero."));
+  }
+};
+
+/*
+ * Generalized Newtonian model data.
+ */
+struct GeneralizedNewtonianModelData
+{
+  GeneralizedNewtonianModelData()
+    : generalized_newtonian_model(GeneralizedNewtonianViscosityModel::Undefined),
+      is_active(false),
+      viscosity_margin(0.0),
+      kappa(0.0),
+      lambda(0.0),
+      a(0.0),
+      n(0.0)
+  {
+  }
+
+  GeneralizedNewtonianViscosityModel generalized_newtonian_model;
```

Comment:
I would expect to find the definition of this enum in the present file. Same for the turbulence model above.

Reply:
and not in the iNS/user_interface/enum_types.h as done now?